README.rst

```diff
@@ -308,6 +308,117 @@ In case you need to run additional instructions just before the build step you c
+              },
+              {
+                /* Insert a custom footer. */
+                op: PLUGIN_OPERATIONS.Insert,
+                widget: {
+                  id: 'custom_footer',
+                  type: DIRECT_PLUGIN,
+                  RenderWidget: () => (
+                    <h1>This is the footer.</h1>
+                  ),
+                },
+              },
+    """
+        )
+    )
+
+Let's take a closer look at what's happening here, starting with the patch name: the prefix is ``mfe-env-config-plugin-`` followed by the name of the slot you're trying to configure. In this case, ``footer_slot``.  (There's a full list of `possible slot names <#mfe-env-config-plugin-slot-catalog>`_ below.) Next, we're hiding the default contents of the footer with a ``PLUGIN_OPERATIONS.Hide``. (Refer to the `frontend-plugin-framework README <https://github.com/openedx/frontend-plugin-framework/#>`_ for a full description of the possible plugin types and operations.) The ``default_contents`` widget ID always refers to what's in an unconfigured slot. Finally, we use ``PLUGIN_OPERATIONS.Insert`` to add our custom JSX component, composed of a simple ``<h1>`` message.  We give it a widgetID of ``custom_footer``.
+
+That's it!  If you rebuild the ``mfe`` image after enabling the plugin, "This is the footer." should appear at the bottom of every MFE.
+
+It's also possible to target a specific MFE's footer. For that, you'd use a patch such as ``mfe-env-config-plugin-profile-footer_slot``, ``profile`` being the name of the MFE and ``footer_slot`` the name of the slot:
+
+.. code-block:: python
+
+    hooks.Filters.ENV_PATCHES.add_item(
+        (
+            "mfe-env-config-plugin-profile-footer_slot",
+            """
+              {
+                /* Hide the global footer we defined earlier. */
+                op: PLUGIN_OPERATIONS.Hide,
+                widgetId: 'custom_footer',
+              },
+              {
+                /* Insert a custom footer specific to the Profile MFE. */
+                op: PLUGIN_OPERATIONS.Insert,
+                widget: {
+                  id: 'custom_profile_footer',
+                  type: DIRECT_PLUGIN,
+                  RenderWidget: () => (
+                    <h1>This is the Profile MFE's footer.</h1>
+                  ),
+                },
+              },
+    """
+        )
+    )
+
+Note that we haven't removed the first ``mfe-env-config-plugin-footer_slot`` patch, so here we hide our ``custom_footer`` instead of ``default_contents``.
+
+If you were to rebuild the MFE image now, the Profile MFE's footer would say "This is the Profile MFE's footer", whereas all the others would still contain the global "This is the footer." message.
+
+.. _mfe-env-config-plugin-slot-catalog:
+
+Here's a list of the currently available frontend plugin slots, separated by category. The documentation of each slot can be found by following the corresponding link.
```

Comment:
What's the strategy to keep this document up to date? Is there a template for new plugin documentation within the MFE that might include a note to update the tutor readme?

Reply:
I figure the strategy can be similar to the one for new feature and waffle flags in edx-platform.  Which is to say, one of two things happens:

* As soon as a new flag/slot is merged in the upstream repo, somebody also opens a PR to Tutor to support it.  This could indeed benefit from some sort of automated reminder, but given that new MFE features are more and more commonly required to work with PR sandboxes as part of the review process, and that PR sandboxes are built with Tutor, adding these to Tutor nightly will simply be a built-in step in getting the slot merged.

* When Tutor and its plugins are updated when a new new release is coming out, part of the work is combing through upstream repos for added and changed features.  This should become less and less necessary as Tutor nightly gets "automatically" updated as described above, but if by any chance a slot or two were missed, this is where they would be added.

Reply:
Sounds good, ty

Reply:
Oh, and one neat thing about this PR is that it will make it possible for PR sandboxes to configure slots for testing (once the Grove plugin is updated to match).

Reply:
This approach has one major drawback: Implementing a new slot will require making an upstream PR to tutor-mfe: this will cause a lot of back-and-forth, especially as we intend to have many more plugin slots in Open edX.

Another minor drawback is the requirement to create a Tutor plugin for users who implement their own plugin slots.

As far as I understand, there is little benefit to having an explicit list of slots here. All it gives us is some syntax sugar. We can achieve a similar result with the following MFE-specific filter:

```
tutormfe.hooks.Filters.PLUGIN_SLOTS.add_item((
            "profile", "footer_slot",
            """
              {
                /* Hide the global footer we defined earlier. */
                op: PLUGIN_OPERATIONS.Hide,
                widgetId: 'custom_footer',
              }"""
))
tutormfe.hooks.Filters.PLUGIN_SLOTS.add_item((
            "profile", "footer_slot",
              """{
                /* Insert a custom footer specific to the Profile MFE. */
                op: PLUGIN_OPERATIONS.Insert,
                widget: {
                  id: 'custom_profile_footer',
                  type: DIRECT_PLUGIN,
                  RenderWidget: () => (
                    <h1>This is the Profile MFE's footer.</h1>
                  )
                }
              }
))"""
```

Reply:
Yes, but listing all the slots gives us one majoir _advantage_: it will be very evident what slots are actually available for a particular version of Tutor.  Sure, we can still document them here anyway, but if we make it so a slot only works if it's explicitly supported, then it's also obligatory to document a slot when it's added here.

In any case, I'll give the PLUGIN_SLOTS filter a shot.  I'm not sure yet if it's going to work with the single `env.config.jsx` approach I tried here.

Reply:
Alright, here's the third take: #240